contracts/src/systems/combat/contracts.cairo

```diff
@@ -0,0 +1,398 @@
+#[dojo::contract]
+mod combat_systems {
+    use eternum::alias::ID;
+
+    use eternum::models::resources::ResourceCost;
+    use eternum::models::position::{Position, HomePosition};
+    use eternum::models::config::SpeedConfig;
+    use eternum::models::movable::{Movable, ArrivalTime};
+    use eternum::models::owner::Owner;
+    use eternum::models::quantity::Quantity;    
+    use eternum::models::combat::{
+        Health, Attack, Defence, Duty, TownWatch
+    };
+
+    use eternum::systems::combat::interface::{
+        ISoldierSystems
+    };
+
+    use eternum::constants::{
+        WORLD_CONFIG_ID, SOLDIER_ENTITY_TYPE
+    };
+
+
+    #[external(v0)]
+    impl SoldierSystemsImpl of ISoldierSystems<ContractState> {
+
+        fn create_soldiers( 
+            self: @ContractState, world: IWorldDispatcher, 
```

Comment:
we should pass in realm_id maybe - and use its location to build

Reply:
the entity_id would be the realm id

Reply:
Since it's only meant to be called by realms, I'll just change the name to realm_id

Reply:
we should try to avoid using realm_id and use realm_entity_id instead in the contracts. Since realm_id already exists in Realm.

There might still be a couple place in contract where we use realm_id but we should refactor